cli/flox-rust-sdk/src/providers/publish.rs

```diff
@@ -36,19 +34,22 @@ pub enum PublishError {
     #[error("The environment is in an unsupported state for publishing")]
     UnsupportEnvironmentState(#[source] Box<dyn error::Error>),
 
+    #[error("There was an error communicating with the catalog")]
+    CatalogError(#[source] Box<dyn error::Error>),
+
     #[error("Could not identify user from authentication info")]
     Unauthenticated,
 }
 
 /// The `Publish` trait describes the high level behavior of publishing a package to a catalog.
 /// Authentication, upload, builds etc, are implementation details of the specific provider.
 /// Modeling the behavior as a trait allows us to swap out the provider, e.g. a mock for testing.
+#[allow(async_fn_in_trait)]
 pub trait Publish {
-    fn publish(
+    async fn publish(
```

Comment:
comment (nb for now): I'm not sure how for up we want to expose the async interfaces. Technically we could cut it short here already using pollster, or use async all the way into the command impl. I think so far we've tried to rather push async down the stack.

Reply:
Made a comment to discuss and revisit.  Thanks.